dvc/remote/s3.py

```diff
@@ -202,16 +204,25 @@ def _list_objects(self, path_info, max_items=None, prefix=None):
             kwargs["Prefix"] = posixpath.join(path_info.path, prefix[:2])
         paginator = self.s3.get_paginator(self.list_objects_api)
         for page in paginator.paginate(**kwargs):
-            yield from page.get("Contents", ())
+            contents = page.get("Contents", ())
+            if progress_callback:
+                progress_callback(len(contents))
+            yield from contents
```

Comment:
```suggestion
            if progress_callback:
                for item in contents:
                    progress_callback()
                    yield item
            else:
	            yield from contents
```

Reply:
I was under the impression that it would be more efficient to update the progress bar once per page rather than per item in each page, especially given that the bar is being updated by multiple threads?

Reply:
no, all the efficiency logic (skipping display etc) is handled by the bar itself.